### PR TITLE
Fixes #5378: removal of trainee delegates from delegate listing

### LINKED ISSUE #5378: Trainee delegates are listed on the Delegates page
According to @cubewhiz in #5118, Trainee Delegates are not (yet) WCA Staff and therefore should not be listed on the Delegate page.
It's probably just a matter of adding a `.reject(&:trainee_delegate?)` [here](https://github.com/thewca/worldcubeassociation.org/blob/master/WcaOnRails/app/views/static_pages/delegates.html.erb#L11).

I noticed this because the the "Trainee Delegate" badge has a link to the Delegates page, so ideally this badge should also be amended to not link there ^^

@danieljames-dj would you mind taking care of this please?

### PR DESCRIPTION
**Pull request by danieljames-dj**

### CODE REVIEW

File: WcaOnRails/app/helpers/persons_helper.rb [hunk 1]
```diff
@@ -33,8 +33,15 @@ def return_podium_class(result)
...
   def delegate_badge(kind)
-    content_tag(:span, class: "badge delegate-badge") do
-      link_to(t("enums.user.delegate_status." + kind), "/delegates", title: t("enums.user.delegate_status." + kind), data: { toggle: "tooltip", placement: "bottom" })
+    title = t("enums.user.delegate_status." + kind)
+    class_list = "badge delegate-badge"
+    data = { toggle: "tooltip", placement: "bottom" }
+    if kind.eql? "trainee_delegate"
+      content_tag(:span, title, class: class_list, data: data)
+    else
+      content_tag(:span, class: class_list) do
```
Comment (danieljames-dj) [inline WcaOnRails/app/helpers/persons_helper.rb:42]: I believe there is some better way to handle this, but I was not able to figure it out. Please let me know if there is a better way. 😬

Comment (viroulep) [reply WcaOnRails/app/helpers/persons_helper.rb:42]: I'm pretty sure you can keep the tooltip on the span, so you can probably do something like that:
```
title = t("enums.user.delegate_status." + kind)
content_tag(:span, class: "badge delegate-badge", data: { toggle: "tooltip", placement: "bottom" }, title: title) do
  kind == "trainee_delegate" ? title : link_to(title, "/delegates")
end
```

**Comment by Ivan-Brigidano** [comment]
I halfway disagree with no displaying Trainee Delegates on the Delegates List. All WAC members are listed on the Committees, Teams and Councils list and only one of them is a WCA Staff Representative (so only one of them is a WCA Staff).

**Comment by viroulep** [comment]
Please discuss this with the Board, I'm just implementing the changes asked [here](https://github.com/thewca/worldcubeassociation.org/issues/5118#issue-557238956).

**Pull request merged**